include/flatbuffers/idl.h

```diff
@@ -420,12 +420,13 @@ struct IDLOptions {
 
 // This encapsulates where the parser is in the current source file.
 struct ParserState {
-  ParserState() : cursor_(nullptr), line_(1), token_(-1) {}
+  ParserState() : cursor_(nullptr), line_(1), token_(-1), is_bool_(false) {}
```

Comment:
That's a bit of a hack :)
The better way is to introduce a `kTokenBooleanConstant`, and to check the uses of `kTokenIntegerConstant` to also deal with this new token (in some cases, not all)